core/src/main/java/org/elasticsearch/index/mapper/NumberFieldMapper.java

```diff
@@ -231,22 +244,39 @@ Query rangeQuery(String field, Object lowerTerm, Object upperTerm,
                 }
                 return fields;
             }
+
+            private void validateParsed(Float value) {
+                if (
+                    value.isNaN() || value.isInfinite()
+                        || value > 65504
```

Comment:
No. Floats that are between 65504 and 65520 will be rounded to 65504 however floats that are equal or greater than 65520 will be converted to +Infinity.